src/marqo/tensor_search/tensor_search.py

```diff
@@ -1746,18 +1750,17 @@ def _vector_text_search(
             for i, q in enumerate(readable_body):
                 if "index" in q:
                     continue
-                for vec in list(q["query"]["nested"]["query"]["knn"].keys()):
-                    readable_body[i]["query"]["nested"]["query"]["knn"][vec]["vector"] = \
-                        readable_body[i]["query"]["nested"]["query"]["knn"][vec]["vector"][:5]
+                try:
+                    for vec in list(q["query"]["nested"]["query"]["knn"].keys()):
+                        if "vector" in q["query"]["nested"]["query"]["knn"][vec]:
+                            readable_body[i]["query"]["nested"]["query"]["knn"][vec]["vector"] = \
+                                readable_body[i]["query"]["nested"]["query"]["knn"][vec]["vector"][:5]
+                except KeyError:
+                    pass
```

Comment:
When does this happen?

Reply:
This will happen when `verbose == 1` and we use a dummy query to replace the original query. 

I am not exactly sure about the best solution here, try except, or do`if "key" in query` check?

Reply:
An `if` check would be much better. Swallowing an exception is bad practice and can cause difficult to debug issues.